### ISSUE TITLE
Error on compile with Ninja

### ISSUE THREAD

**Issue by Hyeokreal** [craigleili/3DLocalMultiViewDesc#1]
Hi Lei Li,

Really thanks for sharing your wonderful works.

I have an issue on compiling the soft_renderer.
When I tried to initialize MultiViewRenderer class, I got following errors.

ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/cpp_extension.py", line 1515, in _run_ninja_build
    env=env)
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "transform.py", line 18, in <module>
    from soft_renderer.cuda.jit import soft_rasterize_cuda as src
  File "/host_temp/codes/cqcpu/point_backends/learnable_viewpoint/soft_renderer/cuda/jit.py", line 9, in <module>
    verbose=True)
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/cpp_extension.py", line 974, in load
    keep_intermediates=keep_intermediates)
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/cpp_extension.py", line 1179, in _jit_compile
    with_cuda=with_cuda)
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/cpp_extension.py", line 1279, in _write_ninja_file_and_build_library
    error_prefix="Error building extension '{}'".format(name))
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/cpp_extension.py", line 1529, in _run_ninja_build
    raise RuntimeError(message)
RuntimeError: Error building extension 'soft_rasterize_cuda'

I tried within both my local machine and a docker container that you provided but in both environment same errors occured.

Do you have any idea on this issue ?

Thanks.

**Comment by Hyeokreal** [comment]
It was a pytorch version issue. Thanks

**Comment by chrisranderson** [comment]
@Hyeokreal What version worked for you?

**Comment by AndreiMiculita** [comment]
This might be useful to other people coming across this. I also couldn't compile the `soft_renderer` at first.

`libcudart.so` was not being found even though I had installed it as per the instructions on nVidia's website. What fixed it was just installing `libtorch3-dev` via apt.

For me it worked with `torch==1.2.0` (as in `requirements.txt`, this was not an issue).

Edit: that was probably not it. It stopped working today for what I can only assume is some nondeterministic reason. I got it working again by switching to a more recent torch version, such that the CUDA version torch is built with matches the version of CUDA installed on the system.
